napari/layers/image/image.py

```diff
@@ -877,27 +872,37 @@ def _on_data_loaded(self, data: ImageSliceData, sync: bool) -> None:
         sync : bool
             If True the chunk was loaded synchronously.
         """
+        # TODO ASYNC: There are quite a few things here which aren't covered by
```

Comment:
I'm unsure if these TODOs are temporary notes or if you want to merge them into `main`.

Personally, I'm not in favor of merging them mostly because there are so many and they're quite repetitive. Instead, I'd prefer to create an issue that tracks that work instead.

Reply:
I actually started this PR by branching from my last one. I had added all those comments to help you understand last week. Now they need to be cleaned up, especially since a lot of these are probably related to the "eventually we need to fix slice state" option that we decided against.

Reply:
Actually some of these raised some important questions regarding what needs to stay in this sequence of calls (the `slice_dims`, `set_view_slice` path). This is going to take some thought and testing to work out.